clients/da-vinci-client/src/main/java/com/linkedin/davinci/blobtransfer/BlobSnapshotManager.java

```diff
@@ -204,6 +214,15 @@ private boolean isSnapshotStale(String topicName, int partitionId) {
         - snapshotTimestamps.get(topicName).get(partitionId) > snapshotRetentionTimeInMillis;
   }
 
+  /**
+   * Increase the count of hosts using the snapshot
+   */
+  void increaseConcurrentUserCount(String topicName, int partitionId) {
```

Comment:
In function: `getConcurrentSnapshotUsers`:
```
 if (!concurrentSnapshotUsers.containsKey(topicName)
        || !concurrentSnapshotUsers.get(topicName).containsKey(partitionId)) {
      return 0;
    }
    return concurrentSnapshotUsers.get(topicName).get(partitionId).get();
```

We can change it to the following to avoid the double map lookups:
```
Map<Integer, AtomicLong> partitionUsageMap = concurrentSnapshotUsers.get(topicName);
if (partitionUsageMap == null) {
  return 0;
}
AtomicLong usage = partitionUsageMap.get(partitionId);
if (usage == null) {
  return 0;
}
return usage.get();
} 
```
In theory `int` should be more than enough to track the usage, right?